generation/WinSDK/emitter.settings.rsp

```diff
@@ -1748,4 +1747,16 @@ WNetUseConnectionA::return=WIN32_ERROR
 WNetUseConnectionW::return=WIN32_ERROR
 HttpReceiveHttpRequest::Overlapped=[Out]
 PFTASKDIALOGCALLBACK::msg=[AssociatedEnum("TASKDIALOG_NOTIFICATIONS")]
-GlobalMemoryStatusEx::lpBuffer=[In]
\ No newline at end of file
+GlobalMemoryStatusEx::lpBuffer=[In]
+DXGI_ADAPTER_DESC1::Flags=[AssociatedEnum("DXGI_ADAPTER_FLAG")]
+DXGI_ADAPTER_DESC2::Flags=[AssociatedEnum("DXGI_ADAPTER_FLAG")]
+DXGI_DECODE_SWAP_CHAIN_DESC::Flags=[AssociatedEnum("DXGI_SWAP_CHAIN_FLAG")]
+DXGI_SWAP_CHAIN_DESC::Flags=[AssociatedEnum("DXGI_SWAP_CHAIN_FLAG")]
+DXGI_SWAP_CHAIN_DESC1::Flags=[AssociatedEnum("DXGI_SWAP_CHAIN_FLAG")]
+IDXGIDevice4::OfferResources1::Flags=[AssociatedEnum("DXGI_OFFER_RESOURCE_FLAGS")]
+IDXGIOutput3::CheckOverlaySupport::pFlags=DXGI_OVERLAY_SUPPORT_FLAG
+IDXGIOutput4::CheckOverlayColorSpaceSupport::pFlags=DXGI_OVERLAY_COLOR_SPACE_SUPPORT_FLAG
+IDXGIOutput6::CheckHardwareCompositionSupport::pFlags=DXGI_HARDWARE_COMPOSITION_SUPPORT_FLAGS
```

Comment:
Don't think these are right. The headers/docs show this as

```
typedef 
enum DXGI_OVERLAY_SUPPORT_FLAG
    {
        DXGI_OVERLAY_SUPPORT_FLAG_DIRECT	= 0x1,
        DXGI_OVERLAY_SUPPORT_FLAG_SCALING	= 0x2
    } 	DXGI_OVERLAY_SUPPORT_FLAG;
```

Which is `int` by default. So it has to stay as-is, however I believe you can associate via the `AssociatedEnum` attribute.

Reply:
See my question above:

> Curiously in the linked PR we had to annotate these via `[AssociatedEnum("…")]`. Same here for struct fields and `UINT`s passed by value, but **not** for `* UINT`s passed as pointer... Is that expected?

If not for the `p` prefix it wouldn't be visible that `pFlags` here is actually a pointer to an enum that we're changing the type of... And there's no complaint that it _needs_ to use `AssociatedEnum`.

Besides making it `AssociatedEnum`, should there be similar validation like https://github.com/microsoft/win32metadata/pull/1583#issuecomment-1561407892?

Reply:
I think you're asking how to use `AssociatedEnum` with pointers (`UINT*` in this case).

@mikebattista Is that something we support?

Reply:
Not really, because I haven't tested it yet. Just asking in the first place if it's expected that the too

Then I assume that it is up to the tooling in e.g. `windows-rs` to understand that a `UINT*` field with `#[AssociatedEnum(XXX)]` annotation really means "a _pointer to_ a value of `XXX`" where `XXX` has the same data layout as `UINT`.

Reply:
It's expected because the metadata tooling doesn't have a lot of tests.

I don't think bindgen on the Rust side is expecting AssociatedEnum on pointers either.

Reply:
Having `AssociatedEnum` on these (out) pointer types breaks windows-rs completely:

https://github.com/Traverse-Research/windows-rs/commit/ea1c32772c4da263b828f2822d0177acd8dadf22

Out values are no longer returned from functions but are passed _by value_ to the function, where _its value_ is cast to a pointer.

Fwiw:

```
Windows.Win32.Graphics.Dxgi.IDXGIOutput3.CheckOverlaySupport : pFlags : [Out] => [AssociatedEnum(DXGI_OVERLAY_SUPPORT_FLAG),Out]
Windows.Win32.Graphics.Dxgi.IDXGIOutput4.CheckOverlayColorSpaceSupport : pFlags : [Out] => [AssociatedEnum(DXGI_OVERLAY_COLOR_SPACE_SUPPORT_FLAG),Out]
Windows.Win32.Graphics.Dxgi.IDXGIOutput6.CheckHardwareCompositionSupport : pFlags : [Out] => [AssociatedEnum(DXGI_HARDWARE_COMPOSITION_SUPPORT_FLAGS),Out]
Windows.Win32.Graphics.Dxgi.IDXGISwapChain3.CheckColorSpaceSupport : pColorSpaceSupport : [Out] => [AssociatedEnum(DXGI_SWAP_CHAIN_COLOR_SPACE_SUPPORT_FLAG),Out]
```

Reply:
@mikebattista @riverar @kennykerr first of all happy new year!

Is there any resolution here? Should we go for the apparently-correct thing which is applying `[AssociatedEnum]` to `UINT *` pointers, and then create an issue over at `windows-rs` to track fixing how it handles this in `bindgen`?

Reply:
Reported at https://github.com/microsoft/windows-rs/issues/2771.

Reply:
@mikebattista Do we want to support adding `AssociatedEnum` attributes to pointers? This would require projection authors to handle these a bit differently for pointers. (That is, ensure the param type is emitted as T* instead of T.)

@AArnott Do you anticipate any issues with that? Rust side will need a small tweak.

Reply:
Thanks for checking. It seems reasonable to me for CsWin32.

Reply:
If it's ok with projections then it's no problem to add to the metadata.

Reply:
The Rust projections requested to ""resolve"" this in the metadata first. At risk of ending up in an infinite loop, it seems like we can just make a choice here and `windows-rs` will follow along with whatever it ends up with.

Reply:
The order of operations is generally: metadata proposes a change, downstream stakeholders provide feedback, metadata makes the change, metadata puts out a release, and then projections adopt or ignore. I don't think it was unreasonable to request, by me, that we wait for something concrete to show up in metadata before doing any work.

Let's move forward with the attribute. You'll still need to correct your emitter.settings.rsp in this PR.

Reply:
It isn't an unreasonable request. It is however unreasonable for downstream stakeholders to send someone right back to the metadata, without feedback on the case.

Reply:
The feedback is gathered here directly and behind the curtain, not via GitHub issues in the projection repositories. I sympathize though, that wasn't clear, thanks for pointing that out!

Reply:
Fair enough, it seemed to me as if we were already with one foot in phase 3: metadata is inclined to make a change. Part of the blame is on me though, the intent of https://github.com/microsoft/windows-rs/issues/2771 was to query downstream projection feedback (as this issue had sat stale for a while - and is a PR thread and not a discussion/issue anyway), though I have not made that abundantly clear.